From 6585262a4dccd78da5cd9adb3437d3e35e3ef775 Mon Sep 17 00:00:00 2001
From: Enis Soztutar <enis@apache.org>
Date: Tue, 5 Apr 2016 18:13:40 -0700
Subject: [PATCH 239/263] HBASE-15587 FSTableDescriptors.getDescriptor() logs
 stack trace erronously

(cherry picked from commit fcfda192f7a9ab4a7950b2dc21cd861abedf49df)

Change-Id: If6c96680f6cd2fafee15b0a4b407c3e8e101042f
Author: Balazs Meszaros
Reason: Bug
Ref: CDH-50427
---
 .../hadoop/hbase/util/FSTableDescriptors.java      |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java
index 7cd2673..06eb9ea 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java
@@ -124,11 +124,13 @@ public class FSTableDescriptors implements TableDescriptors {
     this.metaTableDescriptor = HTableDescriptor.metaTableDescriptor(conf);
   }
 
+  @Override
   public void setCacheOn() throws IOException {
     this.cache.clear();
     this.usecache = true;
   }
 
+  @Override
   public void setCacheOff() throws IOException {
     this.usecache = false;
     this.cache.clear();
@@ -173,6 +175,8 @@ public class FSTableDescriptors implements TableDescriptors {
     } catch (NullPointerException e) {
       LOG.debug("Exception during readTableDecriptor. Current table name = "
           + tablename, e);
+    } catch (TableInfoMissingException e) {
+      // ignore. This is regular operation
     } catch (IOException ioe) {
       LOG.debug("Exception during readTableDecriptor. Current table name = "
           + tablename, ioe);
-- 
1.7.9.5

